### PR TITLE
Rework/cruds

### PR DESCRIPTION
**Pull request by dwaard**
- [x] Room
- [x] Presentation
- [x] Company
- [x] Booths
- [x] Sponsorships
- [x] Users

### CODE REVIEW

File: app/Models/Room.php [hunk 1]
```diff
@@ -53,4 +54,16 @@ public function presentations(): HasMany
     {
         return $this->hasMany(Presentation::class);
     }
+
+    /**
+     * Checks if this room can be deleted.
+     *
+     * @return Attribute
+     */
+    public function canBeDeleted() : Attribute
```
Comment (v-stamenova) [inline app/Models/Room.php:63]: @dwaard do you think we should refactor it in policy?

**Comment by v-stamenova** [comment]
I'm working on Presentation and maybe later on Company in #290. When I am done should I merge them over here instead of merging on development?

**Comment by v-stamenova** [comment]
Before testing: run `sail artisan migrate` because a new table `user_invitation` is created in order to tackle the creation of users from the content moderator

For testing:

Room
- [ ] Create
- [ ] View
- [ ] Edit
- [ ] Delete (all presentations attached to it should get null for room_id and timeslot_id)

Presentation
- [ ] Create 
- [ ] View
- [ ] Edit (should be available to the speaker as well)
- [ ] Delete (check business logic & authorization; all records in `speakers` connected to it are deleted, as well as all connected records from `participants`)

Company
- [ ] Create (two types - one with email invitation, one with already existing user)
- [ ] View
- [ ] Edit (should be available to the company rep as well)
- [ ] Delete (check business logic & authorization; all presentations get deleted therefore all records in `speakers` connected to them are deleted, as well as all connected records from `participants`; all company members get the role simple role 'participant' and are disconnected by the company that gets deleted)

Booth
- [ ] Create
- [ ] View
- [ ] Edit
- [ ] Delete

Sponsorship
- [ ] Create
- [ ] View
- Edit - Not created. Is it needed? 
- [ ] Delete (the `is_sponsor_approved` and `sponsor_tier_id` are null in the `teams`

User
- [ ] Create (via invitation; test the invitation route)
- [ ] View
- Edit - Not created for the company mod, the user themselves have it (jetstream default). Is it needed? 
- [ ] Delete (depends on the role: if participant - removed all connected records in `participants` table; if not-main speaker - removed all connected records in `participants` table, removed record from `speakers`; if main speaker - removed all connected records in `participants` table, and the same things that occur when a presentation is deleted [check above]); if company rep - all things that happen when the company gets deleted + deleting the account of the company rep

**Comment by TimKardol** [comment]
> - [ ] Delete (the `is_sponsor_approved` and `sponsor_tier_id` are null in the `teams`

`sponsor_tier_id` reverts to `null`, `is_sponsor_approved` however reverts to `0` which is different from `null`. This technically shouldn't be any problem.

File: resources/views/emails/user-invite.blade.php
Comment (TimKardol) [inline resources/views/emails/user-invite.blade.php:1]: Text under 'accept your invitation':

You will be registered as a participant by default. When the final program is released you can enrol in any of the presentations using your account.

As soon as your account has been created, you can request to host a presentation. When requesting a presentation make sure to enter the necessary information such as: a summary of the topic which will be shown on the website and the level of difficulty intended for the participants. There are three difficulty levels:

Beginner: No specific knowledge/skill necessary.
Intermediate: Basic level of knowlegde/skill.
Advanced: Experienced level of knowledge/skill.

**Review by TimKardol — CHANGES_REQUESTED** [review]

File: resources/views/moderator/booths/show.blade.php
[the remark this answers is not in the document]
Comment (v-stamenova) [reply resources/views/moderator/booths/show.blade.php:1]: I don't think I actually implemented that one, my bad is it needed?

**Review by TimKardol — APPROVED** [review]

**Pull request merged**